### PR TITLE
[ADAM-1169] Write GFF header line pragma in single file mode.

### LINKED ISSUE #1169: GFF3 formatted features written as single file must include gff-version pragma
"##gff-version 3.2.1

The GFF version follows the format of 3.#.# in this spec. This directive must be present, must be the topmost line of the file. The version number always begins with 3, the second and third numbers are optional and indicate a major revision and a minor revision respectively."

https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1169.

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/FeatureRDD.scala [hunk 1]
```diff
@@ -390,10 +394,18 @@ case class FeatureRDD(rdd: RDD[Feature],
   def saveAsGff3(fileName: String,
                  asSingleFile: Boolean = false,
                  disableFastConcat: Boolean = false) = {
+    val optHeaderPath = if (asSingleFile) {
+      val headerPath = "%s_head".format(fileName)
+      GFFHeaderWriter(headerPath, rdd.context)
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/FeatureRDD.scala]: `GFFHeaderWriter` should be `GFF3HeaderWriter` to match `GFF3InFormatter` and elsewhere.  The GFF2/GTF specifications do not require a version pragma in the header.

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/GFFHeaderWriter.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.adam.rdd.feature
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.SparkContext
+
+/**
+ * Writes the header for a GFF file to an otherwise empty file.
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/GFFHeaderWriter.scala]: GFF &rarr; GFF3

File: adam-core/src/test/scala/org/bdgenomics/adam/rdd/feature/GFFHeaderWriterSuite.scala [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ */
+package org.bdgenomics.adam.rdd.feature
+
+import org.bdgenomics.adam.util.ADAMFunSuite
+import scala.io.Source
+
+class GFFHeaderWriterSuite extends ADAMFunSuite {
+
+  sparkTest("write gff header pragma") {
+    val tmp = tmpFile(".gff")
```
Comment (heuermh) [inline adam-core/src/test/scala/org/bdgenomics/adam/rdd/feature/GFFHeaderWriterSuite.scala]: `".gff"` &rarr; `".gff3"`

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/GFFHeaderWriter.scala [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ */
+private[feature] object GFFHeaderWriter {
+
+  val HEADER_STRING = "##gff-version 3.2.1"
+
+  /**
+   * Writes a GFF Header pragma to a file.
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/GFFHeaderWriter.scala]: GFF → GFF3

**Comment by fnothaft** [comment]
@heuermh addressed comments.

**Comment by heuermh** [comment]
Thank you, @fnothaft!

**Pull request merged**